src/rgw/rgw_rados.h

```diff
@@ -3600,6 +3600,10 @@ class RGWRados
       (get_zonegroup().zones.size() > 1 || current_period.is_multi_zonegroups_with_zones());
   }
 
+  bool can_reshard() const {
+    return zonegroup.zones.size() == 1 && current_period.is_single_zonegroup();
```

Comment:
oops, i tested this with a vstart cluster and it returns false because there is no period. it looks like we check `current_period.get_id().empty()` in other places to see if we have a period or not

Reply:
good catch! 
if there is no period we can always reshard I will fix it